src/odemis/driver/test/xt_client_test.py

```diff
@@ -248,6 +248,20 @@ def test_rotation(self):
         self.assertEqual(self.scanner.rotation.value, init_rotation + 0.01)
         self.scanner.rotation.value = init_rotation
 
+    def test_scan_mode(self):
+        """Test getting and setting the scan mode."""
+        possible_modes = ["external", "full_frame", "spot", "line"]
+        init_scan_mode = self.scanner.scanMode.value
+        self.assertTrue(init_scan_mode in possible_modes)
+        for mode in possible_modes:
+            self.scanner.scanMode.value = mode
+            self.assertEqual(self.microscope.get_scan_mode(), mode)
+
+        with self.assertRaises(IndexError):
+            self.scanner.scanMode.value = "fake_scan_mode"
+
+        self.scanner.scanMode.value = init_scan_mode
```

Comment:
Don't you want to also add the scan_mode and beam_power to the update_settings method to ensure that if these settings are updated on the microscope itself the VA's are also updated?
If yes then this should also be tested in this test case.